jdaviz/configs/specviz/plugins/parsers.py

```diff
@@ -117,28 +117,34 @@ def specviz_spectrum1d_parser(app, data, data_label=None, format=None, show_in_v
                 elif i == 0:
                     app.add_data_to_viewer("spectrum-viewer", data_label[i])
 
-        # reset display ranges, or build combined spectrum, when input is a SpectrumList instance
         if isinstance(data, SpectrumList):
-
-            # build combined spectrum
-            wlallarr = np.array(wlallorig)
-            fnuallarr = np.array(fnuallorig)
-            dfnuallarr = np.array(dfnuallorig)
-            srtind = np.argsort(wlallarr)
-            wlall = wlallarr[srtind]
-            fnuall = fnuallarr[srtind]
-            fnuallerr = dfnuallarr[srtind]
-
-            # units are not being handled properly yet.
-            unc = StdDevUncertainty(fnuallerr * flux_units)
-            spec = Spectrum1D(flux=fnuall * flux_units, spectral_axis=wlall * wave_units,
-                              uncertainty=unc)
-
-            # Make metadata layout conform with other viz.
-            spec.meta = standardize_metadata(spec.meta)
-
-            # needs perhaps a better way to label the combined spectrum
-            label = "Combined " + data_label[0]
-            app.add_data(spec, label)
-            if show_in_viewer:
-                app.add_data_to_viewer("spectrum-viewer", label)
+            if data[0].meta.get("INSTRUME") == "MIRI":
+                # reset display ranges, or build combined spectrum,
+                # when input is a SpectrumList instance and the
+                # observations were produced by MIRI
+                wlallarr = np.array(wlallorig)
+                fnuallarr = np.array(fnuallorig)
+                dfnuallarr = np.array(dfnuallorig)
+                srtind = np.argsort(wlallarr)
+                wlall = wlallarr[srtind]
+                fnuall = fnuallarr[srtind]
+                fnuallerr = dfnuallarr[srtind]
+
+                # units are not being handled properly yet.
+                unc = StdDevUncertainty(fnuallerr * flux_units)
+                spec = Spectrum1D(flux=fnuall * flux_units, spectral_axis=wlall * wave_units,
+                                  uncertainty=unc)
+
+                # Make metadata layout conform with other viz.
+                spec.meta = standardize_metadata(spec.meta)
+
+                # needs perhaps a better way to label the combined spectrum
+                label = "Combined " + data_label[0]
+                app.add_data(spec, label)
+                if show_in_viewer:
+                    app.add_data_to_viewer("spectrum-viewer", label)
+            else:
+                # show only the first FITS extension if the observations
+                # were produced by instrument other than MIRI
+                if show_in_viewer:
+                    app.add_data_to_viewer("spectrum-viewer", data_label[0])
```

Comment:
does this block need an `add.add_data call` or is that handled somewhere else?

Reply:
Pretty sure that's done [up here](https://github.com/spacetelescope/jdaviz/blob/75eb05f3e48feb653df5c93f2fe4f71d4056bcc6/jdaviz/configs/specviz/plugins/parsers.py#L105).

Reply:
Exactly! 👍

Reply:
Is this necessary? `add_data_to_viewer` is already done in L118 above.

Reply:
My understanding was that you still needed to call `add_data` before you call `add_data_to_viewer` – is that right?

Reply:
Yes, I think so...